File: Henry_example_SQLITE3_C/example_4_SQLITE3_Create_Table_2.c
/*  http://zetcode.com/DB/sqlitec/
Compile:
$ gcc example_SQLITE3_program_name.c -o out -l sqlite3

Execute:
$ ./out
*/

#include <sqlite3.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    sqlite3 *DB;
    char *messaggeError = 0; // created error message

/************************************************************/
/* sqlite3_libversion(): Returns the SQLite library version */
/************************************************************/
   printf("sql version = %s\n", sqlite3_libversion()); 

/************************************************************/
/*                     Open Database File                   */
/************************************************************/
    int exit = sqlite3_open("example.db", &DB);
   
// If exit is an integer, exit is equal to 0, it succeeded. If exit is equal to 1, database doesn’t exist yet. The database is going to be created in the same directory where the process was executed.
   if( exit ) {
      fprintf(stderr, "Cannot open database: %s\n", sqlite3_errmsg(DB)); // sqlite3_errmsg() function returns a description of the error.
      sqlite3_close(DB);
      return 1;  
   } else {
      fprintf(stderr, "Opened Database Successfully\n");
   }

/************************************************************/
/*             Creat data table - Create SQL statement      */
/************************************************************/ 
// We connect to the example.db database, and create the table.  
    char *sql = "CREATE TABLE Cars("
                 "Id INT PRIMARY KEY     NOT NULL,"
                 "Name     TEXT          NOT NULL,"
                 "Price    INT           NOT NULL);";

// Execute SQL commands provided by sql argument which can consist of more than one SQL command.
    exit = sqlite3_exec(DB, sql, 0, 0, &messaggeError);

// SQLITE_OK is returned when the connection was successfully established. 
    if (exit != SQLITE_OK ) {
        fprintf(stderr, "SQL error: %s\n", messaggeError);
        sqlite3_free(messaggeError);   // The allocated message string must be freed with the sqlite3_free() function call.     
        sqlite3_close(DB);
        return 1;
    } else {
        fprintf(stdout, "Table created Successfully:\n");
    }

// closes the database connection.
    sqlite3_close(DB);
    
    return 0;
}

File: Henry_example_SQLITE3_PYTHON/example_4_SQLITE3_Create_Table_2.py
#!/usr/bin/python
import sqlite3 as lite
from sqlite3 import Error
import sys 


def create_connection(db_file):
    """ create a database connection to the SQLite database
        specified by db_file
    :param db_file: database file
    :return: Connection object or None
    """
    try:
        conn = lite.connect(db_file)
        return conn
    except Error as e:
        print(e)
        sys.exit(1)
    return None

def create_table(conn):
    """ create a table from the create_table_sql statement
    :param conn: Connection object
    :return:
    """
    try:
        cursor = conn.cursor()
        cursor.execute('''CREATE TABLE COMPANY
                     (Id INT PRIMARY KEY     NOT NULL,
                      Name           TEXT    NOT NULL,
                      Price            INT     NOT NULL);''')

        print "Table created successfully";

    except Error as e:
        print(e)
        sys.exit(1)

if __name__ == '__main__':
    conn = create_connection("example.db")
    create_table(conn)
    conn.close()
